src/assets/CurrencyRateController.ts

```diff
@@ -128,9 +128,10 @@ export class CurrencyRateController extends BaseController<CurrencyRateConfig, C
 	 */
 	async fetchExchangeRate(currency: string, nativeCurrency = this.activeNativeCurrency): Promise<CurrencyRateState> {
 		const json = await handleFetch(this.getPricingURL(currency, nativeCurrency));
+		const exchange: string | undefined = json[currency.toUpperCase()];
 		return {
 			conversionDate: Date.now() / 1000,
-			conversionRate: Number(json[currency.toUpperCase()]),
+			conversionRate: exchange === undefined ? 0 : Number(exchange),
```

Comment:
We'd might as well update this to be a bit more robust, e.g. confirm that the value is indeed a number. This could still return `NaN`, if given an empty string (which I know that API probably doesn't do, but still)

Reply:
The _Convert numeric strings and null to numbers_ section has the following, non-number results:

```
Number('foo')     // NaN
Number('100a')    // NaN
Number('-Infinity') //-Infinity
```

And we know that:

```
> Number(undefined)
NaN
```

Your point stands that there are cases where we _could_ get `NaN`, but none of those seem, to me, like reasonable things to guard against. Guarding against the property not existing seems okay, but  guarding that we don't get a incoherent string or `Infinity`/`-Infinity` seems unreasonable—there's no evidence that those are possible values?

Reply:
I mean... it's a third party API. They could decide to start returning a poop emoji and we couldn't do anything to stop them 🤷‍♂ . Isn't it reasonable to check for any possibility, especially when we can easily do so with a single condition?

Reply:
Oh wait, the `undefined` check has to come before using the `Number` constructor, because that returns zero. Hmm.

Then I guess it'd have to be two checks - one for nullish, then a call to `Number.isFinite` to check the _output_ of the `Number` constructor? I think that'd do it 🤔